bishop.cpp

```diff
@@ -0,0 +1,60 @@
+#include "bishop.h"
+
```

Comment:
Since you forward declare `Piece`, include it in the cpp file.

```suggestion
#include "piece.h"
```